### PR TITLE
Improve example `textinput`

### PR DESCRIPTION
**Pull request by blankRiot96**
Fixes issue <a href="https://github.com/pygame/pygame/issues/3040">#3040</a>. 
I was unable to add the cursor blink due to personal time constraints, but I've refactored the code and re structured it to use OOP.

### CODE REVIEW

**Comment by dr0id** [comment]
I don't think that it is intended that the first written text gets behind the input box?
image

File: examples/textinput.py [hunk 1]
```diff
@@ -6,169 +6,235 @@
...
+    def __init__(
+        self, promt: str,
+        pos, screen_dimensions,
+        print_event: bool,
+        text_color="white"
+    ) -> None:
+        self.promt = promt
+        self.print_event = print_event
+        # position of chatlist and chatbox
+        self.CHAT_LIST_POS = pg.Rect(pos, (screen_dimensions[0], 400))
```
Comment (dr0id) [inline examples/textinput.py:46]: Use a slightly different position for the CHAT_LIST_POS (+50 in y direction would do the trick I guess)

**Comment by blankRiot96** [comment]
Weird it works fine for me 
image

File: examples/textinput.py (hunk 1 above)
Comment (blankRiot96) [reply examples/textinput.py:46]: Got it, thanks for the review!

**Comment by dr0id** [comment]
> Weird it works fine for me image

Well, yes I can input 'normal text' fine but I thought I try some special utf characters.
Thanks for the position fix.

**Review by dr0id — APPROVED** [review]

**Comment by blankRiot96** [comment]
Thanks for the approval! My first time here so I wasn't too sure if I was doing it right 😅

**Comment by dr0id** [comment]
@robertpfeiffer did you say in discord that you had it working with UTF symbols too? I wonder what is needed to get it working with UTF symbols. Can you elaborate? I do not mind if it is not in this PR.

**Comment by robertpfeiffer** [comment]
> UTF

I had written a textinput example app in 2020. We discussed it on Discord. I did not add it to the pygame examples because it doesn't work in any *meaningful* way with UTF-8 without something like GNU Unifont (that covers the whole Unicode basic multilingual plane). It also doesn't work properly with ibus/mozc or fcitx, because of the way manylinux wheels are linked. So I had written that textinput example that works really well, but more or less only on my machine, because of packaging issues.

The same issues are also affecting this textinput example. It's not as nice looking and feature-rich as the thing I posted on the Discord, but that's not really the problem. The problem is it's not properly working when you install pygame from the official wheels.

I guess that means I am giving my :+1: for this PR. The issue is packaging though.

**Comment by dr0id** [comment]
> The issue is packaging though.

Thanks for the explanation. In this case I would say this PR is good to be done.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Pull request merged**